# Question:1

# The time required for servicing transmissions is normally distributed with  = 45 minutes and  = 8 minutes. The service manager plans to have work begin on the transmission of a customer’s car 10 minutes after the car is dropped off and the customer is told that the car will be ready within 1 hour from drop-off. What is the probability that the service manager cannot meet his commitment? 

In [5]:
# import required libraries
from scipy import stats
from scipy.stats import norm

In [6]:
# Here the service manager plans to finish the work within 50 minutes of the car is dropped off
# Find Z-Scores at X=50; Z = (X - µ) / σ 
Z=(50-45)/8
Z

0.625

In [7]:
# The probability that the service manager cannot meet his requirement is time is greater than 50 minutes
# Find probability P(X>50) = 1-stats.norm.cdf(abs(z_score))
1-stats.norm.cdf(abs(0.625))

0.26598552904870054

# The probability that the service manager cannot meet his requirement is approximately 26.5% 

#

#

# Question:2

# The current age (in years) of 400 clerical employees at an insurance claims processing center is normally distributed with mean  = 38 and Standard deviation  =6. For each statement below, please specify True/False. If false, briefly explain why.


# More employees at the processing center are older than 44 than between 38 and 44.
# A training program for employees under the age of 30 at the center would be expected to attract about 36 employees.


In [8]:
# QA.More employees at the processing center are older than 44 than between 38 and 44.

In [9]:
# find the probability of employees older tahn 44 years
# p(X>44); Employees older than 44 yrs of age
1-stats.norm.cdf(44,loc=38,scale=6)

0.15865525393145707

In [10]:
# Fine the probability taht the employees are between age of 38 and 44
# p(38<X<44); Employees between 38 to 44 yrs of age
stats.norm.cdf(44,38,6)-stats.norm.cdf(38,38,6)

0.3413447460685429

# From the above we can say that  More employees at the processing center are older than 44 than between 38 and 44 is a false statement

In [11]:
# QB: A training program for employees under the age of 30 at the center would be expected to attract about 36 employees
# Find the probability that the employees are aged under 30 years
# P(X<30); Employees under 30 yrs of age
stats.norm.cdf(30,38,6)

0.09121121972586788

# The probability that the employees are under age of 30 is approximately 9% , Which is 36 employees are aged below 30years

# A training program for employees under the age of 30 at the center would be expected to attract about 36 employees is a True statement

#

# Question:3  Let X ~ N(100, 202). Find two values, a and b, symmetric about the mean, such that the probability of the random variable taking a value between them is 0.99. 

In [12]:
# Here mean = 100 and standard deviation = 20
# here we have to calculate the 99% confidence interval that the random variable is taking
stats.norm.interval(0.99,100,20)

(48.48341392902199, 151.516586070978)

# The answer is = (48.48341392902199, 151.516586070978)

#

Consider a company that has two different divisions. The annual profits from the two divisions are independent and have distributions Profit1 ~ N(5, 3^2) and Profit2 ~ N(7,4^2 ) respectively. Both the profits are in Dollar Million. Answer the following questions about the total profit of the company in Rupees. Assume that Dollar = Rs. 45
Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.
Specify the 5th percentile of profit (in Rupees) for the company
Which of the two divisions has a larger probability of making a loss in a given year?

In [13]:
# the company is having profits from two dofferent divisions
# Mean profits from two different divisions of a company = Mean1 + Mean2 and one dollar is equal to 45 rupees
Mean = 5+7
print('Mean Profit is Rs', Mean*45,'Million')

Mean Profit is Rs 540 Million


In [14]:
# Variance of profits from two different divisions of a company = SD^2 = SD1^2 + SD2^2
import numpy as np
SD = np.sqrt((9)+(16))
print('Standard Deviation is Rs', SD*45, 'Million')

Standard Deviation is Rs 225.0 Million


In [15]:
# Q.A: Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.
print('Range is Rs',(norm.interval(0.95,540,225)),'in Millions')

Range is Rs (99.00810347848784, 980.9918965215122) in Millions


In [19]:
#Q.B: Specify the 5th percentile of profit (in Rupees) for the company
# To compute 5th Percentile, we use the formula X=μ + Zσ; wherein from z table, 5 percentile = -1.645
X= 540+(-1.645)*(225)
print('5th percentile of profit  is',np.round(X,2),'Million Rupees')

5th percentile of profit  is 169.88 Million Rupees


In [22]:
# Q.C:Which of the two divisions has a larger probability of making a loss in a given year?
# Probability of Division 1 making a loss P(X<0)
print('Probability of Division 1 making a loss is',stats.norm.cdf(0,5,3))
print('Probability of Division 2 making a loss is',stats.norm.cdf(0,7,4))

Probability of Division 1 making a loss is 0.0477903522728147
Probability of Division 2 making a loss is 0.040059156863817086


# From the above  we can say that Division 1 has slightly higher propability of 0.007 of making a loss when compared to Division2